### PR TITLE
Copy all exercises to alternate tree, build these

### PR DESCRIPTION
**Pull request by KevinWMatthews**
Avoid renaming the original example files to keep a clean git diff.
The alternate tree is ignored - it's only used to build.

### CODE REVIEW

**Comment by KevinWMatthews** [comment]
@patricksjackson  @elyashiv Proof of concept to ameliorate #226 . Does it work for you?

Pros:
  * Original `exercises/` directory is untouched
  * New `build_exercises/` directory is built and ignored by git
  * Works with in-tree and out-of-tree builds

Cons:
  * If the examples are changed, must re-run CMake to recopy the exercises into the alternate directory

Nitpicks:
  * I'm hard-coding `alt_exercise_tree` into `travis_fixup()`. Should probably add an extra argument.
  * What should we name stuff?

I'm looking at having the recopy run with every build - I think that facilitate a convenient workflow. There are a few sticking points with CMake semantics, so I'll come back to it if you like this.

**Comment by KevinWMatthews** [comment]
Took a stab at fixing my own nitpicks.

**Comment by KevinWMatthews** [comment]
@patricksjackson You're right; this is an unconventional workflow that deserves some explanation. I'll try it out - don't mean to shirk writing documentation - but you're welcome to edit heavily.

**Comment by KevinWMatthews** [comment]
Ok, there's a first draft. I added in some background information; I figure it's easy to remove if it's too verbose.

**Comment by KevinWMatthews** [comment]
I still hope that there's a way to make CMake a bit smarter. It may be possible to `add_custom_command()` to each target and have this custom command copy only the files for that particular exercise. That would remove a user step and provide an extra bit of granularity.

It might get complicated, however, so if you like the current CMake system as-is I'm inclined to start with this and make a separate pull request later.

File: README.md [hunk 1]
```diff
@@ -22,3 +22,35 @@ Here is a list of things that need to be done to add an exercise to this track.
...
+The current CMake build system navigates this with an unusual approach: it copies
+all example solutions from the `exercises/` tree to an alternate directory
+(which is ignored by git), renames the solutions as if they were student exercises,
+and runs a complete build in this new directory.
+
+Maintainers can largely ignore the alternate exercise directory if they recopy
+their example solution before running a build. Re-running CMake will recopy
+all exercise files.
+
+For example, a maintainer can copy, configure, compile, and test all exercises with:
```
Comment (patricksjackson) [inline README.md]:
```suggestion
For example, a maintainer can copy, configure, compile, and test all exercises by
running the following from the root directory:
```

File: README.md [hunk 2]
```diff
@@ -22,3 +22,35 @@ Here is a list of things that need to be done to add an exercise to this track.
...
+```bash
+cmake . && make
+```
+
+For an individual exercise, in this example `anagram`, the maintainer should edit:
+```bash
+exercises/anagram/example.h
+exercises/anagram/example.cpp
```
Comment (patricksjackson) [inline README.md]:
```suggestion
exercises/anagram/example.cpp
exercises/anagram/anagram_test.cpp
```

**Comment by patricksjackson** [comment]
I have a few small nits , but otherwise is LGTM. (Just push yourself after fixing if those are fine).

This is already pretty simple, I'm not sure how it's an extra step unless you mean you could somehow get `make test_anagram` to do the copy and then build and run the tests?

**Comment by KevinWMatthews** [comment]
Good nits. Trying out the GitHub interface for resolving/committing changes.

Yes, exactly that. It seems possible, at least in principle, to to set up a custom command or target that will copy the files for just the `anagram` executable. If the copy command/target could be configured as always out of date and as a prerequisite for the `anagram` target, then the dependency should trickle through to `test_anagram` and would be run every time.

The trick is that this can't be visible to students so it has to happen in the root `CMakeLists.txt` file. It might be an unreasonable amount of work.

**Pull request merged**